src/cascadia/TerminalSettingsEditor/AISettings.xaml

```diff
@@ -126,23 +134,106 @@
+                            </StackPanel>
+                        </Grid>
+                        <Grid RowSpacing="8"
+                              Visibility="{x:Bind mtu:Converters.InvertedBooleanToVisibility(ViewModel.IsOpenAIKeySet), Mode=OneWay}">
+                            <Grid.RowDefinitions>
+                                <RowDefinition Height="Auto" />
+                                <RowDefinition Height="Auto" />
+                                <RowDefinition Height="Auto" />
+                                <RowDefinition Height="Auto" />
+                                <RowDefinition Height="Auto" />
+                                <RowDefinition Height="Auto" />
+                            </Grid.RowDefinitions>
+                            <StackPanel Grid.Row="0"
+                                        Padding="16"
+                                        CornerRadius="4"
+                                        Orientation="Horizontal"
+                                        Spacing="14">
+                                <StackPanel.Background>
+                                    <SolidColorBrush Opacity="0.7"
+                                                     Color="#404040" />
+                                </StackPanel.Background>
+                                <muxc:InfoBadge x:Name="OpenAIDescriptionInfoBadge"
+                                                VerticalAlignment="Top"
+                                                FontSize="16"
+                                                Style="{ThemeResource AttentionIconInfoBadgeStyle}">
+                                    <muxc:InfoBadge.IconSource>
+                                        <muxc:FontIconSource FontSize="16"
+                                                             Glyph="&#xE625;" />
+                                    </muxc:InfoBadge.IconSource>
+                                </muxc:InfoBadge>
+                                <RichTextBlock TextWrapping="WrapWholeWords"
+                                               MaxWidth="900">
+                                    <Paragraph>
+                                        <Run x:Name="AISettings_OpenAIDescriptionPart1" /><Hyperlink NavigateUri="https://go.microsoft.com/fwlink/?linkid=2251839"
+                                                   TextDecorations="None">
+                                            <Run x:Name="AISettings_OpenAIDescriptionLinkText" />
+                                        </Hyperlink><Run x:Name="AISettings_OpenAIDescriptionPart2" />
+                                    </Paragraph>
+                                </RichTextBlock>
+                            </StackPanel>
+                            <TextBlock x:Uid="AISettings_KeyTextBox"
+                                       Grid.Row="3"
+                                       VerticalAlignment="Center" />
+                            <TextBox x:Name="OpenAIKeyInputBox"

```

Comment:
Could you remind me if this (and the AOAI counterpart) could be PasswordBox controls instead?

Reply:
Yep they can! Updated